### ISSUE TITLE
allow developer to select desired implementation of stdin, etc.

### ISSUE THREAD

**Issue by gneverov** [raspberrypi/pico-sdk#1900]
[picolibc_interface.c](https://github.com/raspberrypi/pico-sdk/blob/develop/src/rp2_common/pico_clib_interface/picolibc_interface.c) currently defines strong symbols for stdin, stdout, and stderr that use the implementation from [pico_stdio](https://github.com/raspberrypi/pico-sdk/tree/develop/src/rp2_common/pico_stdio). There is no way for the developer to override this behavior.

### Use cases
- if picolibc was compiled with POSIX_IO, then picolibc already defines weak versions of stdin, etc.. Probably the developer intends to use this implementation, but it is silently overridden by the pico-sdk's implementation, which uses strong symbols.
- the developer might define their own symbols for stdin, etc. and doesn't want them overridden or in conflict with what pico-sdk defines.

**Comment by vazhnov** [comment]
Is it fixed by https://github.com/raspberrypi/pico-sdk/pull/2935 ?

**Comment by gneverov** [comment]
I'm not sure. I resolved my original issue by creating my own "interface.c" rather than relying on the built-in ones. I'm fine with closing this issue.

**Comment by vazhnov** [comment]
@gneverov , before to close the issue, you can share your recipe with others :)

**Comment by kilograham** [comment]
yeah, i think, leave this open, as it is a reasonable enhancement

**Comment by gneverov** [comment]
@vazhnov, @kilograham mentioned at the time that the C lib interface system is user extensible. You can define the CMake PICO_CLIB variable to point to your own implementation instead of using a built-in one.
